### DISCUSSION TITLE
filter history by author

### DISCUSSION THREAD

**Discussion by nikhedonia** [community/community#24426] [Apps, API and Webhooks]
<p>I’m trying to fetch all commits of one author for a given repository.</p>
<pre><code class="lang-auto">user(login: "nikhedonia") {  
&amp;nbsp; databaseId # =&amp;nbsp;4670055  
}  
repository(owner: "loopperfect", name: "buckaroo") {  
&amp;nbsp; nameWithOwner  
&amp;nbsp; defaultBranchRef {  
&amp;nbsp; &amp;nbsp; name  
&amp;nbsp; &amp;nbsp; target {  
&amp;nbsp; &amp;nbsp; ... on Commit {  
&amp;nbsp; &amp;nbsp; &amp;nbsp; history(author: {id:4670055}) {  
&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; totalCount  
&amp;nbsp; &amp;nbsp; &amp;nbsp; }  
&amp;nbsp; &amp;nbsp; }  
&amp;nbsp; }  
}  
</code></pre>
<p>Unfortunately I recieve the following error:</p>
<pre><code class="lang-auto">"errors": [  
{  
"type": "NOT\_FOUND",  
"path": [  
"repository",  
"defaultBranchRef",  
"target",  
"history",  
"totalCount"  
],  
"locations": [  
{  
"line": 18,  
"column": 13  
}  
],  
"message": "Could not resolve to a node with the global id of '4670055'"  
}  
]  
</code></pre>
<p>What ID am I supposed to use?</p>
<p>btw. Google Indexed this link: <a href="https://platform.github.community/t/filter-repository-history-by-login-instead-of-id/4592%C2%A0but" rel="nofollow noopener">https://platform.github.community/t/filter-repository-history-by-login-instead-of-id/4592 but</a> it seems to be out of date</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/nikhedonia">@nikhedonia</a>,</p>
<p>Thanks for being here! That link from google is for the old platform board, it has since migrated here you can read more about it <a href="https://github.community/t/welcome-to-the-github-api-development-and-support-board/13591">here</a>.</p>
<p>Here is a sample query that might help:</p>
<pre><code>{
  repository(name: "andreagriffiths11.me", owner: "andreagriffiths11") {
    ref(qualifiedName: "master") {
      target {
        ... on Commit {
          id
          history(first: 100) {
            pageInfo {
              hasNextPage
            }
            edges {
              node {
                messageHeadline
                oid
                message
                author {
                  name
                  email
                  date
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by nikhedonia** [comment]
<p>Thanks for your quick reply.</p>
<p>Your example iterates through the first 100 commits.</p>
<p>The docs mention that you can filter the commits by <strong>author_id</strong> and/or <strong>author_email</strong>. </p>
<p>Unfortunately all my attempts of filtering by <strong>author_id </strong> caused a fatal error:</p>
<p>&gt; Could not resolve to a node with the global id of ‘4670055’</p>
<p>This does not seem right to me.</p>
<p>&gt; That link from google is for the old platform board</p>
<p>Are the old posts still accesible?</p>
<p>The link <a href="https://platform.github.community/t/filter-repository-history-by-login-instead-of-id/4592%C2%A0but" rel="nofollow noopener">https://platform.github.community/t/filter-repository-history-by-login-instead-of-id/4592</a> and googles teaser seem to be realated to my problem.</p>
<p>Unfortunately I was not able to find the original post neither using google or the the searchfunction of this community forum</p>

**Comment by sorenlouv** [comment, marked as answer]
<p><a class="mention" href="/nikhedonia">@nikhedonia</a> It looks like you are currently trying to filter by a databaseId. Instead you should filter by the user id. You can get the user id if you know the username like:</p>
<pre><code>{
  user(login: "nikhedonia") {
    id # MDQ6VXNlcjQ2NzAwNTU=
  }
}
</code></pre>
<p>The user id for the username “nikhedonia” is “MDQ6VXNlcjQ2NzAwNTU=”. You can now filter commits by user id:</p>
<pre><code>history(first: 5, author: {id: "MDQ6VXNlcjQ2NzAwNTU="}) {
</code></pre>
